Sources/ApodiniGraphQL/GraphQLInterfaceExporter.swift

```diff
@@ -0,0 +1,150 @@
+            default:
+                fatalError("\(error)")
+            }
+        } catch {
+            fatalError("\(error)")
+        }
+    }
+    
+    func export<H: Handler>(blob endpoint: Endpoint<H>) where H.Response.Content == Blob {
+        logger.error("Blob endpoint \(endpoint) cannot be exported")
+    }
+    
+    func finishedExporting(_ webService: WebServiceModel) {
+        let schema: GraphQLSchema
+        do {
+            schema = try schemaBuilder.finalize()
+        } catch {
+            fatalError("Error finalizing GraphQL schema: \(error)")
+        }
+        app.httpServer.registerRoute(.GET, config.graphQLEndpoint, responder: GraphQLQueryHTTPResponder(schema: schema))
+        app.httpServer.registerRoute(.POST, config.graphQLEndpoint, responder: GraphQLQueryHTTPResponder(schema: schema))
+        if config.enableGraphiQL {
+            registerGraphiQLEndpoint()
+        }
+    }
+    
+    private func registerGraphiQLEndpoint() {
+        let graphqlEndpointUrl = app.httpConfiguration.uriPrefix + config.graphQLEndpoint.httpPathString
+        app.httpServer.registerRoute(.GET, config.graphiQLEndpoint) { req -> HTTPResponse in
+            guard let url = Bundle.module.url(forResource: "graphiql", withExtension: "html") else {
+                throw HTTPAbortError(status: .internalServerError)
+            }
+            guard var htmlPage = (try? Data(contentsOf: url)).flatMap({ String(data: $0, encoding: .utf8) }) else {
+                throw HTTPAbortError(status: .internalServerError)
+            }
+            htmlPage = htmlPage.replacingOccurrences(of: "{{APODINI_GRAPHQL_ENDPOINT_URL}}", with: graphqlEndpointUrl)
+            return HTTPResponse(
+                version: req.version,
+                status: .ok,
+                headers: HTTPHeaders {
+                    $0[.contentType] = .html
+                },
+                bodyStorage: .buffer(.init(string: htmlPage))
+            )
+        }
+    }
+}
+
+
+struct GraphQLEndpointDecodingStrategy: EndpointDecodingStrategy {
+    typealias Input = Map
+    
+    func strategy<Element: Codable>(for parameter: EndpointParameter<Element>) -> AnyParameterDecodingStrategy<Element, Input> {
+        GraphQLEndpointParameterDecodingStrategy<Element>(name: parameter.name).typeErased
+    }
+}
+
+
+private struct GraphQLEndpointParameterDecodingStrategy<T: Codable>: ParameterDecodingStrategy {
+    typealias Element = T
+    typealias Input = Map
+    
+    struct Error: Swift.Error {
+        let message: String
+    }
+    
+    private struct Wrapped<T: Codable>: Codable {
+        let value: T
+    }
+    
+    let name: String
+    
+    func decode(from input: Input) throws -> T {
+        guard let value = try input.dictionaryValue(converting: true)[name] else {
+            throw Error(message: "Unable to find parameter named '\(name)' (T: \(T.self))")
```

Comment:
I would suggest to thrown an `ApodiniError` instead of a custom error struct here.

Reply:
I intentionally didn't use the `ApodiniError` type because I thought it was only for errors thrown in handlers? (at least that's what it says in the documentation)

Reply:
I would say that `ApodiniError`s should be used everywhere whey they might be returned to the client, so they should be everywhere when they are part of the request/response pipeline including the handler execution but also including the encoding/decoding. This is also especially relevant for #405 and #410 after that.
Would also be good to update that quote in the documentation there. The Docs are more user focused but this is also an information that would be useful for internal development.

Reply:
Done.

Reply:
Thanks! 💪